dbms/src/Interpreters/executeQuery.cpp

```diff
@@ -231,6 +233,11 @@ std::tuple<ASTPtr, BlockIO> executeQueryImpl(
             context.setProcessListElement(&process_list_entry->get());
         }
 
+        // Do set-up work for tunnels and receivers after ProcessListEntry is constructed,
+        // so that we can propagate current_memory_tracker into them.
+        context.getDAGContext()->tunnel_set->updateMemTracker();
```

Comment:
better to check if `getDAGContext` returns nullptr

Reply:
good idea, I wonder when will it be nullptr?

Reply:
sometimes we may run query using TiFlash client, dag context will be nullptr in this case.

Reply:
got it